app/controllers/spree/admin/base_controller_decorator.rb

```diff
@@ -61,7 +61,7 @@ def model_class
 
   def active_distributors_not_ready_for_checkout
     ocs = OrderCycle.managed_by(spree_current_user).active
-    distributors = ocs.map(&:distributors).flatten.uniq
+    distributors = ocs.includes(:distributors).map(&:distributors).flatten.uniq
     Enterprise.where('enterprises.id IN (?)', distributors).not_ready_for_checkout
   end
```

Comment:
A better improvement would be to totally refactor `active_distributors_not_ready_for_checkout` so that we get the enterprise ids with a complex but single query that avoids round trips to the DB but it is not as low-hanging-fruit as this first solution.

Reply:
Another quick win for Ruby performance and Rails 4 compatibility:

```rb
    distributor_ids = ocs.includes(:distributors).map(&:distributor_ids).flatten.uniq
    Enterprise.where('enterprises.id IN (?)', distributor_ids).not_ready_for_checkout
```

Reply:
I prefer to separate these two things because:

* This is already an improvement so I don't want to delay it to get both #iterative